shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h

```diff
@@ -107,6 +110,13 @@ NS_ASSUME_NONNULL_BEGIN
              (void (^)(UNNotificationPresentationOptions options))completionHandler
     API_AVAILABLE(ios(10));
 
+/**
+ * Calls all plugins registered for `UNUserNotificationCenterDelegate` callbacks.
+ */
+- (void)userNotificationCenter:(UNUserNotificationCenter*)center
```

Comment:
The `FlutterAppLifeCycleProvider` now conforms to `UNUserNotificationCenterDelegate`. So `FlutterPlugin` should require this now, right?

Reply:
FlutterAppDelegate is calling   
```
[_lifeCycleDelegate userNotificationCenter:center
              didReceiveNotificationResponse:response
                       withCompletionHandler:completionHandler];
```
and _lifeCycleDelegate is a `FlutterPluginAppLifeCycleDelegate`, so that's the protocol that needs the method either by declaring it `@optional` or by conforming to `UNUserNotificationCenterDelegate`.

Instead of explicitly declaring `-userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler API_AVAILABLE(ios(10))` in the headers anywhere I would just have those protocols and/or classes conform to UNUserNotificationCenterDelegate so you just get whatever methods the SDK thinks you should (unlikely example: they add a new required method to that protocol.  Now you get nice compilation errors that you are missing required implementation of those methods).

Up to you though!